Don't panic when user omits part of command line

<!-- This form is for bug reports and feature requests ONLY!
-->

**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
>
> /kind feature


**What happened**:
Ran 
ks param  set num_ps 1
but meant to run
ks param set myjob num_ps 1

and got:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xfc181e]

goroutine 1 [running]:
github.com/ksonnet/ksonnet/pkg/actions.(*ParamSet).setLocal(0xc420344900, 0xc4203965d0, 0x1, 0x1, 0x1141840, 0xc4200b37b0, 0x1, 0x7ffe1aa2333e)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/actions/param_set.go:134 +0x6e
github.com/ksonnet/ksonnet/pkg/actions.(*ParamSet).Run(0xc420344900, 0xc420344900, 0x0)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/actions/param_set.go:112 +0x22f
github.com/ksonnet/ksonnet/pkg/actions.RunParamSet(0xc4201aa030, 0xc4200abc80, 0x14)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/actions/param_set.go:36 +0x5c
github.com/ksonnet/ksonnet/pkg/clicmd.runAction(0x14, 0xc4201aa030, 0x132fc96, 0x9)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/clicmd/actions.go:113 +0x66
github.com/ksonnet/ksonnet/pkg/clicmd.glob..func24(0x20f63e0, 0xc42000a180, 0x2, 0x2, 0x0, 0x0)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/clicmd/param_set.go:59 +0x3de
github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra.(*Command).execute(0x20f63e0, 0xc42000a100, 0x2, 0x2, 0x20f63e0, 0xc42000a100)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra/command.go:753 +0x468
github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x20f89e0, 0x13, 0xff, 0xc4200500f0)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra/command.go:843 +0x30a
github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra.(*Command).Execute(0x20f89e0, 0xc4200a4058, 0x0)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra/command.go:791 +0x2b
main.main()
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/cmd/ks/main.go:35 +0x85
d

**What you expected to happen**:

An error telling me I omitted a parameter

**How to reproduce it (as minimally and precisely as possible)**:
Given a param set, run
ks param  set num_ps 1


**Anything else we need to know?**:

**Environment**:
- ksonnet version (use `ks version`):
$ ks version
ksonnet version: 0.10.2
jsonnet version: v0.10.0
client-go version: 1.8

- Kubernetes version (use `kubectl version`):
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.2", GitCommit:"81753b10df112992bf51bbc2c2f85208aad78335", GitTreeState:"clean", BuildDate:"2018-04-27T09:22:21Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.2-gke.1", GitCommit:"75d2af854b1df023c7ce10a8795b85d3dd1f8d37", GitTreeState:"clean", BuildDate:"2018-05-10T17:23:18Z", GoVersion:"go1.9.3b4", Compiler:"gc", Platform:"linux/amd64"}
d

--- a/pkg/actions/param_set.go
+++ b/pkg/actions/param_set.go
@@ -131,6 +131,10 @@ func (ps *ParamSet) setLocal(path []string, value interface{}) error {
 		return errors.Wrap(err, "could not find component")
 	}
 
+	if c == nil {
+		return errors.Errorf("unable to find component %s", ps.rawPath)
+	}
+
 	if err := c.SetParam(path, value); err != nil {
 		return errors.Wrap(err, "set param")
 	}

--- a/pkg/actions/param_set_test.go
+++ b/pkg/actions/param_set_test.go
@@ -54,6 +54,31 @@ func TestParamSet(t *testing.T) {
 	})
 }
 
+func TestParamSet_invalid_component(t *testing.T) {
+	withApp(t, func(appMock *amocks.App) {
+		componentName := "deployment"
+		path := "replicas"
+		value := "3"
+
+		in := map[string]interface{}{
+			OptionApp:   appMock,
+			OptionName:  componentName,
+			OptionPath:  path,
+			OptionValue: value,
+		}
+
+		a, err := NewParamSet(in)
+		require.NoError(t, err)
+
+		a.resolvePathFn = func(app.App, string) (component.Module, component.Component, error) {
+			return nil, nil, nil
+		}
+
+		err = a.Run()
+		require.Error(t, err)
+	})
+}
+
 func TestParamSet_asString(t *testing.T) {
 	withApp(t, func(appMock *amocks.App) {
 		componentName := "deployment"

